aspnetcore/grpc/configuration.md

```diff
@@ -18,31 +18,50 @@ The following table describes options for configuring gRPC services:
-| `ResponseCompressionAlgorithm` | `null` | The compression algorithm used to compress messages sent from the server. The algorithm must match a compression provider in `CompressionProviders`. For the algorthm to compress a response the client must indicate it supports the algorithm by sending it in the **grpc-accept-encoding** header. |
+| `ResponseCompressionAlgorithm` | `null` | The compression algorithm used to compress messages sent from the server. The algorithm must match a compression provider in `CompressionProviders`. For the algorithm to compress a response, the client must indicate it supports the algorithm by sending it in the **grpc-accept-encoding** header. |
 | `ResponseCompressionLevel` | `null` | The compress level used to compress messages sent from the server. |
 
 Options can be configured for all services by providing an options delegate to the `AddGrpc` call in `Startup.ConfigureServices`.
 
-```csharp
-public void ConfigureServices(IServiceCollection services)
-{
-    services.AddGrpc(options =>
-    {
-        options.EnableDetailedErrors = true;
-    });
-}
-```
+[!code-csharp[](~/grpc/configuration/sample/GrcpService/Startup.cs?name=snippet)]
 
 Options for a single service override the global options provided in `AddGrpc` and can be configured using `AddServiceOptions<TService>`:
 
-```csharp
-services.AddGrpc().AddServiceOptions<MyService>(options =>
-{
-    options.ReceiveMaxMessageSize = 10 * 1024 * 1024; // 10 megabytes
-});
-```
+[!code-csharp[](~/grpc/configuration/sample/GrcpService/Startup2.cs?name=snippet)]
+
+## Configure Kestrel options
+
+Kestrel server has configuration options that affect the behavior of gRPC on ASP.NET Core.
+
+### Request body data rate limit
+
+The default Kestrel [minimum request body data rate](
+<xref:Microsoft.AspNetCore.Server.Kestrel.Core.KestrelServerLimits.MinRequestBodyDataRate>) is 240 bytes/second. For client streaming and duplex streaming calls, this rate may not be satisfied and the connection may be timed out. The minimum request body data rate limit should typically be disabled when the gRPC service includes client streaming and duplex streaming calls. The following code disables Kestrel minimum request body data rate:
+
+[!code-csharp[](~/grpc/configuration/sample/GrcpService/Program.cs?name=snippet&highlight=13-16)]
+
+## Configure client and server maximum send and receive options
+
+The following table shows the client `Channel` maximum send and receive options and corresponding server options:
+
+|ChannelOptions (Client) |  GrpcServiceOptions (Server) |
+| ------ | ------------- |
+| `MaxSendMessageLength` | `ReceiveMaxMessageSize` |
+| `MaxReceiveMessageLength` | `SendMaxMessageSize` |
+
+<!-- What are important User Agent options ? 
+And some client specific configurations like 
+ -->
+
+The following highlighted code sets the server options listed in the preceding table:
+
+[!code-csharp[](~/grpc/configuration/sample/GrcpService/Startup2.cs?name=snippet)]
```

Comment:
Setting the options in `AddServiceOptions<MyService>` will only change the limits for that service (`MyService`). Setting them in AddGrpc will change them for all services (unless overridden).

IMO change this to set these values in AddGrpc and add a link to AddServiceOption and explain about also changing them per service.